runtime/apps/apis/dma/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+// Licensed under the Apache-2.0 license
+
+//! # AsyncDMA: A DMA Interface for AXI Source to AXI Destination Transfers
+//!
+//! This library provides an abstraction for performing asynchronous Direct Memory Access (DMA)
+//! transfers between AXI source and AXI destination addresses.
+
+use libtock_platform::{ErrorCode, Syscalls};
+use libtockasync::TockSubscribe;
+
+/// DMA interface.
+pub struct AsyncDMA<const DRIVER_NUM: u32, S: Syscalls>(S);
```

Comment:
I think we can make this `DRIVER_NUM` just a constant in the library. I don't think we need it to be a generic parameter.